homeassistant/components/upnp/__init__.py

```diff
@@ -191,6 +194,15 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
     return await hass.config_entries.async_unload_platforms(config_entry, PLATFORMS)
 
 
+@dataclass
+class UpnpSensorEntityDescription(
+    SensorEntityDescription, BinarySensorEntityDescription
+):
+    """A class that describes UPnP entities."""
```

Comment:
These do not mix.

Leave this as a generic `UpnpEntityDescription`, without any inherits, and use the correct base for new ones for each platform.

Reply:
Sorry, missed notification on this and hadn't come back and checked till today.

Wasn't 100% sure how to do this, so split it into 2 data classes. One for sensor and 1 for binary sensor.